include/ginkgo/core/base/executor.hpp

```diff
@@ -724,6 +818,27 @@ class Executor : public log::EnableLogging<Executor> {
 
 #undef GKO_ENABLE_VERIFY_MEMORY_TO
 
+    /**
+     * Populates the executor specific info from the global machine topology
+     * object.
+     *
+     * @param mach_topo the machine topology object.
+     */
+    virtual void populate_exec_info(const MachineTopology *mach_topo) = 0;
+
+    /**
+     * Gets the modifiable exec info object
+     *
+     * @return  the pointer to the exec_info object
+     */
+    exec_info &get_exec_info()
+    {
+        return const_cast<exec_info &>(
+            static_cast<const Executor &>(*this).get_exec_info());
```

Comment:
Though this is ugly, I think this is valid code. Actually it is something recommended by Scott Meyers in his book (Effective C++).  Here is what he says: 

> Casting away the const on the return value is safe, in this case,
because whoever called the non- const getter must have had a non-
const object in the first place. Otherwise they couldn’t have called a
non- const function. So having the non- const getter call the const
version is a safe way to avoid code duplication, even though it requires
a cast. 

Moreover this is a protected member and hence accesible only within this class and its derived classes.

Reply:
I don't understand why you go through so much casts for the simple line:
`return exec_info_;`
`get_exec_info()` only works for non-const `*this`, therefore, you don't need the `const_cast`.
Additionally, having 2 consecutive `const_cast`s does not make sense. You only use `const_cast` to remove const-ness, meaning the inner-cast is used in a wrong manner (if anything, it should be casted with a `static_cast`).

Reply:
You are right. I am not really sure what I was thinking here. Sorry for the confusion.